From 0de82acab34900535d0c15a929f1963fc93f49e0 Mon Sep 17 00:00:00 2001
From: Yi Liu <yi.l.liu@intel.com>
Date: Mon, 22 Nov 2021 17:52:34 +0800
Subject: [PATCH 069/141] vfio_ap: Use vfio_alloc/put_device()

Uses vfio_alloc/put_device() for the ap_matrix_mdev allocation, and no
release callback is needed.

Signed-off-by: Yi Liu <yi.l.liu@intel.com>

4	7	drivers/s390/crypto/vfio_ap_ops.c

diff --git a/drivers/s390/crypto/vfio_ap_ops.c b/drivers/s390/crypto/vfio_ap_ops.c
index 4c62ba9c72d8..0932ef466502 100644
--- a/drivers/s390/crypto/vfio_ap_ops.c
+++ b/drivers/s390/crypto/vfio_ap_ops.c
@@ -462,11 +462,10 @@ static int vfio_ap_mdev_probe(struct mdev_device *mdev)
 	struct ap_matrix_mdev *matrix_mdev;
 	int ret;
 
-	matrix_mdev = kzalloc(sizeof(*matrix_mdev), GFP_KERNEL);
+	matrix_mdev = vfio_alloc_device(ap_matrix_mdev, vdev,
+					&mdev->dev, &vfio_ap_matrix_dev_ops);
 	if (!matrix_mdev)
 		return -ENOMEM;
-	vfio_init_group_dev(&matrix_mdev->vdev, &mdev->dev,
-			    &vfio_ap_matrix_dev_ops);
 
 	matrix_mdev->mdev = mdev;
 	vfio_ap_matrix_init(&matrix_dev->info, &matrix_mdev->matrix);
@@ -485,8 +484,7 @@ static int vfio_ap_mdev_probe(struct mdev_device *mdev)
 	mutex_lock(&matrix_dev->lock);
 	list_del(&matrix_mdev->node);
 	mutex_unlock(&matrix_dev->lock);
-	vfio_uninit_group_dev(&matrix_mdev->vdev);
-	kfree(matrix_mdev);
+	vfio_put_device(&matrix_mdev->vdev);
 	return ret;
 }
 
@@ -500,8 +498,7 @@ static void vfio_ap_mdev_remove(struct mdev_device *mdev)
 	vfio_ap_mdev_reset_queues(matrix_mdev);
 	list_del(&matrix_mdev->node);
 	mutex_unlock(&matrix_dev->lock);
-	vfio_uninit_group_dev(&matrix_mdev->vdev);
-	kfree(matrix_mdev);
+	vfio_put_device(&matrix_mdev->vdev);
 }
 
 static ssize_t name_show(struct mdev_type *mtype,
-- 
2.34.1

